### PR TITLE
Support Slack interactive messages request

### PR DESCRIPTION
**Pull request by tw0517tw**
#24
Added support to Slack interactive message events.
Added an example for Slack interactive message events.
Modified some type annotations.
Changed some server to make sure they response empty body.
Ensured every server support `x-www-form-urlencoded` body parsing.

Tested functionality with the example with 4 servers.

Not yet done:
- [x] add tests for new features
- update document website

### CODE REVIEW

**Comment by CentaurWarchief** [comment]
awesome @tw0517tw — I'm 100% sure that I owe you a beer! How do I pay?

🍻

File: examples/slack-interactive-messge/index.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+const bot = new SlackBot({
+  accessToken: '__FILL_YOUR_TOKEN_HERE__',
+});
+
+bot.onEvent(async context => {
+  if (context.event.isText && context.event.text === 'heyo') {
+    // send a message with buttons and menu
+    await context.postMessage(undefined, {
+      attachments: JSON.stringify([
```
Comment (tw0517tw) [inline examples/slack-interactive-messge/index.js]: `stringify` isn't needed here, underlying client did will do the trick

File: src/bot/SlackConnector.js [hunk 2]
```diff
@@ -15,17 +20,24 @@ export type SlackUser = {
...
+  type: EventAPITypes,
+  event: Message,
   authed_users: Array<string>,
   event_id: string,
   event_time: number,
 };
 
+export type SlackRequestBody = EventsAPIBody | { payload: string };
+
+export type SlackEventSource =
```
Comment (tw0517tw) [inline src/bot/SlackConnector.js]: this is not needed

**Comment by CentaurWarchief** [comment]
@chentsulin is it published?

**Comment by chentsulin** [comment]
Not yet, but we will cut a new release tomorrow. 😉

**Comment by CentaurWarchief** [comment]
@chentsulin you rock. Thanks buddy!

**Pull request merged**